### PR TITLE
Integrate sphinx for deployment

### PR DESCRIPTION
**Pull request by smol-ninja**
- Adds `runBroadcast` and `runSphinx` to support deployments via both forge script and sphinx plugin.
- Sphinx Deployment command has been updated on Warp drive under "Deploy via Sphinx" folder.
- Rename env `ETH_FROM` to `EOA` in line with Sphinx configurations.

### CODE REVIEW

**Comment by smol-ninja** [comment]
@andreivladbrg can you please review this PR?

File: script/Base.s.sol [hunk 1]
```diff
@@ -53,10 +54,13 @@ contract BaseScript is Script {
         }
     }
 
-    modifier broadcast() {
-        vm.startBroadcast(broadcaster);
-        _;
-        vm.stopBroadcast();
+    function configureSphinx() public override {
```
Comment (andreivladbrg) [inline script/Base.s.sol:57]: let's add natspec for this function, referencing sphinx docs/repo

**Review by andreivladbrg — COMMENTED** [review]
generally looks good, but i couldn't test it on my own since i don't have access (yet) to sphinx

what if we keep the `broadcast` modifier, and support both types of deployments?

<details>
<summary> Deploy Core script example </summary>

```solidity
function runSphinx(address initialAdmin)
        public
        virtual
        sphinx
        returns (
            SablierV2LockupDynamic lockupDynamic,
            SablierV2LockupLinear lockupLinear,
            SablierV2LockupTranched lockupTranched,
            SablierV2NFTDescriptor nftDescriptor
        )
    {
        (lockupDynamic, lockupLinear, lockupTranched, nftDescriptor) = _run(initialAdmin);
    }

    function runBroadcast(address initialAdmin)
        public
        virtual
        broadcast
        returns (
            SablierV2LockupDynamic lockupDynamic,
            SablierV2LockupLinear lockupLinear,
            SablierV2LockupTranched lockupTranched,
            SablierV2NFTDescriptor nftDescriptor
        )
    {
        (lockupDynamic, lockupLinear, lockupTranched, nftDescriptor) = _run(initialAdmin);
    }

    function _run(address initialAdmin)
        internal
        returns (
            SablierV2LockupDynamic lockupDynamic,
            SablierV2LockupLinear lockupLinear,
            SablierV2LockupTranched lockupTranched,
            SablierV2NFTDescriptor nftDescriptor
        )
    {
        nftDescriptor = new SablierV2NFTDescriptor();
        lockupDynamic = new SablierV2LockupDynamic(initialAdmin, nftDescriptor, maxCount);
        lockupLinear = new SablierV2LockupLinear(initialAdmin, nftDescriptor);
        lockupTranched = new SablierV2LockupTranched(initialAdmin, nftDescriptor, maxCount);
    }

```

</details>

so we run the CLIs:
- forge script script/DeployCore.s.sol --rpc-url sepolia --sig "runBroadcast(address)" $ADMIN
- bun sphinx propose script/DeployCore.s.sol --networks testnets --sig "runSphinx(address)" $ADMIN

File: script/Base.s.sol (hunk 1 above)
Comment (smol-ninja) [reply script/Base.s.sol:57]: > what if we keep the broadcast modifier, and support both types of deployments?

I am with you on this. Until Sphinx becomes really stable, we should keep both the options.

**Review by andreivladbrg — APPROVED** [review]
Note: we should update warp drive accordingly

**Comment by smol-ninja** [comment]
@andreivladbrg can you please have a final look at the latest commit? If good, I will merge it.

**Review by andreivladbrg — APPROVED** [review]

**Comment by PaulRBerg** [comment]
just came here to say that I love the fact that Warp Drive gets mentioned in our GitHub PRs

Such a [useful feature](https://twitter.com/PaulRBerg/status/1670725786679099392)!

**Pull request merged**